File: Control Hub FTC-5EO4_11-22-2024-17.22.12.981.revbkp/blocks/mecanum encoder.js
// IDENTIFIERS_USED=colorSensorAsColorSensor,gamepad1,mtrBlAsDcMotor,mtrBrAsDcMotor,mtrFlAsDcMotor,mtrFrAsDcMotor

/**
 * Describe this function...
 */
function colorTelementry() {
  if (1 == colorSensorAsColorSensor.getBlue()) {
    telemetry.addLine('colorSensor: Blue');
  } else if (1 == colorSensorAsColorSensor.getRed()) {
    telemetry.addLine('colorSensor: Red');
  } else {
    telemetry.addLine('colorSensor: None');
  }
}

/**
 * Describe this function...
 */
function mecanumSetup() {
  mtrFrAsDcMotor.setMode("RUN_USING_ENCODER");
  mtrBrAsDcMotor.setMode("RUN_USING_ENCODER");
  mtrBlAsDcMotor.setMode("RUN_USING_ENCODER");
  mtrBrAsDcMotor.setMode("RUN_USING_ENCODER");
}

/**
 * This function is executed when this OpMode is selected from the Driver Station.
 */
function runOpMode() {
  linearOpMode.waitForStart();
  telemetry.addLine('waitForStart');
  mecanumSetup();
  if (linearOpMode.opModeIsActive()) {
    telemetry.addLine('opModelsActive');
    while (linearOpMode.opModeIsActive()) {
      mecanumMove();
      gamepadTelem();
      mtrPosTelem();
      colorTelementry();
      telemetry.update();
    }
  }
}

/**
 * Describe this function...
 */
function mtrPosTelem() {
  telemetry.addTextData('mtrFr', String(mtrFrAsDcMotor.getCurrentPosition()));
  telemetry.addTextData('mtrBr', String(mtrBrAsDcMotor.getCurrentPosition()));
  telemetry.addTextData('mtrBl', String(mtrBlAsDcMotor.getCurrentPosition()));
  telemetry.addTextData('mtrFl', String(mtrFlAsDcMotor.getCurrentPosition()));
}

/**
 * Describe this function...
 */
function mecanumMove() {
  mtrBlAsDcMotor.setDualVelocity(gamepad1.getRightStickY() * -1, mtrBrAsDcMotor, gamepad1.getRightStickY());
}

/**
 * Describe this function...
 */
function gamepadTelem() {
  telemetry.addTextData('GP1: RightStickX', String(gamepad1.getRightStickX()));
  telemetry.addTextData('GP1: RightStickY', String(gamepad1.getRightStickY()));
  telemetry.addTextData('GP1: LeftStickX', String(gamepad1.getLeftStickX()));
  telemetry.addTextData('GP1: LeftStickY', String(gamepad1.getLeftStickY()));
}

File: Control Hub FTC-5EO4_11-23-2024-9.54.15.334.revbkp/java/src/org/firstinspires/ftc/teamcode/mecanumencoder.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
import com.qualcomm.robotcore.hardware.DcMotor;

@TeleOp(name = "mecanumencoder (Blocks to Java)")
public class mecanumencoder extends LinearOpMode {

  private DcMotor mtrFr;
  private DcMotor mtrBr;
  private DcMotor mtrBl;
  private DcMotor mtrFl;

  /**
   * This function is executed when this OpMode is selected from the Driver Station.
   */
  @Override
  public void runOpMode() {
    mtrFr = hardwareMap.get(DcMotor.class, "mtrFr");
    mtrBr = hardwareMap.get(DcMotor.class, "mtrBr");
    mtrBl = hardwareMap.get(DcMotor.class, "mtrBl");
    mtrFl = hardwareMap.get(DcMotor.class, "mtrFl");

    // Put initialization blocks here.
    waitForStart();
    if (opModeIsActive()) {
      // Put run blocks here.
      while (opModeIsActive()) {
        // Put loop blocks here.
        mtrPosTelem();
        telemetry.update();
      }
    }
  }

  /**
   * Describe this function...
   */
  private void mtrPosTelem() {
    telemetry.addData("mtrFr", mtrFr.getCurrentPosition());
    telemetry.addData("mtrBr", mtrBr.getCurrentPosition());
    telemetry.addData("mtrBl", mtrBl.getCurrentPosition());
    telemetry.addData("mtrFl", mtrFl.getCurrentPosition());
  }
}
